### PR TITLE
Action check kernel config version

### PR DESCRIPTION
**Pull request by amshinde**
github-action: Add step to verify kernel config version id updated
    
The version mentioned in the `kata_config_version` needs to be
updated for any kernel config change or changed to the patches applied.
Without this, CI would not test with the latest kernel changes.
We use to enforce this earlier as part of CI when `packaging` was
a standalone repo.
Add back this check as part of a github action so that the check is
performed early on instead of a CI job.

Fixes: kata-containers/tests#5419

### CODE REVIEW

**Review by cmaf — APPROVED** [review]
thanks @amshinde

**Review by GabyCT — APPROVED** [review]

File: .github/workflows/static-checks.yaml [hunk 1]
```diff
@@ -38,6 +37,22 @@ jobs:
         go-version: 1.19.3
       env:
         GOPATH: ${{ runner.workspace }}/kata-containers
+    - name: Check kernel config version
+      run: |
+        cd "${{ github.workspace }}/src/github.com/${{ github.repository }}"
+        kernel_dir="tools/packaging/kernel/"
+        kernel_version_file="${kernel_dir}kata_config_version"
+        modified_files=$(git diff --name-only origin/main..HEAD)
+        result=$(git whatchanged origin/main..HEAD "${kernel_dir}" >>"/dev/null")
```
Comment (liubin) [inline .github/workflows/static-checks.yaml:46]: Nit: var `result` is not used.

**Pull request merged**